ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -207,46 +204,29 @@ def get_tables_name_and_type(self) -> Optional[Iterable[Tuple[str, str]]]:
             if self.source_config.includeTables:
                 for table_name in self.inspector.get_table_names(schema_name):
                     table_name = self.standardize_table_name(schema_name, table_name)
-                    table_fqn = fqn.build(
-                        self.metadata,
-                        entity_type=Table,
-                        service_name=self.context.database_service.name.__root__,
-                        database_name=self.context.database.name.__root__,
-                        schema_name=self.context.database_schema.name.__root__,
-                        table_name=table_name,
-                    )
                     if filter_by_table(
-                        self.source_config.tableFilterPattern, table_fqn=table_fqn
+                        metadata=self.metadata,
+                        table_filter_pattern=self.source_config.tableFilterPattern,
+                        context=self.context,
+                        table_name=table_name,
+                        status=self.status,
+                        use_fqn=self.source_config.useFqnForFiltering,
                     ):
-                        self.status.filter(
-                            table_fqn,
-                            "Table pattern not allowed",
-                        )
                         continue
-
                     yield table_name, TableType.Regular
 
             if self.source_config.includeViews:
                 for view_name in self.inspector.get_view_names(schema_name):
                     view_name = self.standardize_table_name(schema_name, view_name)
-                    view_fqn = fqn.build(
-                        self.metadata,
-                        entity_type=Table,
-                        service_name=self.context.database_service.name.__root__,
-                        database_name=self.context.database.name.__root__,
-                        schema_name=self.context.database_schema.name.__root__,
-                        table_name=view_name,
-                    )
-
                     if filter_by_table(
-                        self.source_config.tableFilterPattern, table_fqn=view_fqn
+                        metadata=self.metadata,
+                        table_filter_pattern=self.source_config.tableFilterPattern,
+                        context=self.context,
+                        table_name=view_name,
+                        status=self.status,
+                        use_fqn=self.source_config.useFqnForFiltering,
```

Comment:
what about having a `Filter` class that we do at the beginning:
```python
filter = Filter(metadata=self.metadata,
                      table_filter_pattern=self.source_config.tableFilterPattern,
                      context=self.context,
                      status=self.status,
                      use_fqn=self.source_config.useFqnForFiltering)
```
and then we call it using just `filter.table_matches(table_name=view_name)`. Wdyt?